jme3-alloc/src/main/java/com/jme3/alloc/util/NativeBufferUtils.java

```diff
@@ -0,0 +1,116 @@
+package com.jme3.alloc.util;
+
+import java.io.IOException;
+import java.nio.Buffer;
+import com.jme3.alloc.util.loader.NativeImageLoader;
+
+/**
+ * A jvm low-level implementation to the native GNU-Standard-Library [stdlib].
+ * 
+ * @author pavl_g
+ */
+public final class NativeBufferUtils {
+
+    static {
+        try {
+            /* extracts and loads the system specific library */
+            NativeImageLoader.loadLibrary();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private NativeBufferUtils() {
+    }
+
+    /**
+     * Allocates a buffer of size bytes.
+     * 
+     * @param size the size of the newly created buffer in bytes (max = 32-bit or 4 bytes)
+     * @return a new java-nio buffer object
+     */
+    public static native Buffer memoryAlloc(final long size);
+
+    /**
+     * Copies the value [value] into each of the first [size] bytes of the object beginning at the 
+     * memory address of [buffer].
+     * 
+     * @param buffer a buffer block to start from
+     * @param value the value to set the memory blocks with starting from the buffer
+     * @param size the size to extend after the buffer address
+     * @return the buffer block
+     */
+    public static native void memorySet(final Buffer buffer, final int value, final long size); 
+
+    /**
+     * 
+     * @param to
+     * @param from
+     * @param size
+     * @return
+     */
+    public static native void memoryMove(final Buffer to, final Buffer from, final long size);
```

Comment:
I wonder why this method is needed, can't we do this from java?

Reply:
I am not sure if these will work on java nio buffers: 
1) `memoryMove`
2) `memorySet`
3) `realloc`
4) `reallocarray`

Reply:
I will test them today, and we should decide here before merging this PR.

Reply:
Removed `realloc` and `reallocarray`, since the `java.nio.Buffer` has a constant capacity value, so those have an undefined behavior, while `memorySet` and `memoryMove` are working fine, i will provide techdemos for them....